src/server/worker/driver/driver.go

```diff
@@ -361,24 +361,16 @@ func (d *driver) RunUserCode(
 	if err != nil {
 		return errors.EnsureStack(err)
 	}
-	// A context with a deadline will successfully cancel/kill
-	// the running process (minus zombies)
-	state, err := cmd.Process.Wait()
-	if err != nil {
-		return errors.EnsureStack(err)
-	}
-	if common.IsDone(ctx) {
-		if err = ctx.Err(); err != nil {
+	select {
```

Comment:
I feel like Wait should respect the context and not require a quick check here to see if it's done.  What if it's done right after the chan recv fails and is done right before Wait runs?  Seems pointless to check.